app/services/ZuoraService.scala

```diff
@@ -27,8 +30,69 @@ class ZuoraService(zuoraApiConfig: ZuoraApiConfig) extends ZuoraApi {
   override val metrics = new ZuoraMetrics(stage, application)
   val authTask = ScheduledTask(s"Zuora ${apiConfig.envName} auth", Authentication("", ""), 0.seconds, 30.minutes)(request(Login(apiConfig)))
 
-  def createSubscription(memberId: MemberId, data: SubscriptionData, ratePlanId: String, paymentDelay: Option[Period]): Future[SubscribeResult] = {
-    request(Subscribe(memberId, data, ratePlanId, paymentDelay))
+  def createSubscription(memberId: MemberId, data: SubscriptionData, paymentDelay: Option[Period]): Future[SubscribeResult] = {
+    request(Subscribe(memberId, data, paymentDelay))
   }
+
+  private def getProducts(): Future[Seq[SubscriptionProduct]] = {
+
+    def productRatePlans: Future[Seq[ProductRatePlan]] =
+      query[ProductRatePlan](s"ProductId='$digitalProductId'")
+
+
```

Comment:
Do we need to execute several HTTP calls to Zuora to extract a list of `SubscriptionProduct`? Is there a way to do this in a single task? Perhaps it is worth spending some time figuring out a better way of doing this using the Zuora query language